pkg/bsql/helpers.go

```diff
@@ -134,3 +140,25 @@ func parseTimeWithEngine(value string, dbEngine database.DbEngine) (*time.Time,
 	// Fall back to the generic parser if prioritized formats don't match
 	return parseTime(value)
 }
+
+// generateCredentials generates a random password based on the credential options and configuration.
+func generateCredentials(credentialOptions *v2.CredentialOptions) (string, error) {
+	if credentialOptions == nil || credentialOptions.GetRandomPassword() == nil {
+		return "", errors.New("unsupported credential option: only random password is supported")
+	}
+
+	randomPasswordOpts := credentialOptions.GetRandomPassword()
+
+	if randomPasswordOpts.GetLength() < PasswordMinLength {
+		randomPasswordOpts = &v2.CredentialOptions_RandomPassword{
+			Length:      PasswordMinLength,
```

Comment:
Minimum password length is already set to 8 in the proto https://github.com/ConductorOne/baton-sdk/blob/main/proto/c1/connector/v2/resource.proto#L153 and is enforced in GenerateRandomPassword: https://github.com/ConductorOne/baton-sdk/blob/main/pkg/crypto/password.go#L59

If we want minimum password lengths of 12, we should probably change that in baton-sdk, not just in one connector. Also if we add more fields to CredentialOptions_RandomPassword in the future, this code will fail to copy them.